docs/docs/service-discovery-load-balancing.md

```diff
@@ -6,15 +6,37 @@ title: Service Discovery & Load Balancing
-
-## Ports Assignment
-
-When an application is created in Marathon (either through the REST API or the front end) - you may assign it one or more ports. These ports may be any valid port number or 0, which Marathon will use to randomly assign a port number.
-
-This port is used to ensure no two applications can be run using Marathon with overlapping port assignments (i.e. that the ports are globally unique).
-
-However, this is not the port that each instance of that application is assigned when running on slave nodes within the Mesos cluster. Since multiple instances may run on the same node, each instance (or *task*) is assigned a random port. This can be accessed from the `$PORT` environment variable which is set by Marathon.
+An alternative way to implement service discovery is to run a TCP proxy on each host in the cluster, and transparently forward 
+connections to the static service port on localhost to the dynamically assigned host/port combinations of the individual 
+app instances (= *tasks*). That way, clients simply connect to that port, and the implementation details of discovery 
+are completely abstracted away. This approach is sufficient if all apps are launched through Marathon. 
+
+## Service Ports Assignment
+
+When Marathon creates an application in Marathon (either through the REST API or the front end),
+you may assign one or more service ports to it. 
+You can specify an valid port numbers as service ports or you can use 0 to indicate that Marathon should assign
+service ports. 
+
+For applications using docker and `BRIDGE` networking, you assign the servicePorts in the 
+`portMappings`. For applications with `HOST` networking or without docker, you simply use the `ports` attribute
+to assign the service ports. Look at the [REST API docs]({{ site.baseurl }}/docs/rest-api.html#ports-array-of-integers)
+for details.
+
+Marathon will ensure that all dynamically assigned service ports are unique for this Marathon framework instance.
+If you run multiple Marathon frameworks on the cluster, you should make sure that their ranges for the service
+ports do not overlap. You can configure this via the `--local_port_min` and `--local_port_max` 
+[command line flags](({{ site.baseurl }}/docs/command-line-flags.html)).
+Marathon does not use these assigned service ports itself. 
+It is the responsibility of a separate component to make use of this information and set up load balancing accordingly. 
+A sample script that does this is provided with Marathon, the `haproxy-marathon-bridge`.
+
+Note that these service ports should not be confused with the host ports of the individual tasks. 
+Since multiple tasks may run on the same node, host ports are usually assigned dynamically per task if
+you do not use `"requirePorts"` or set the `"hostPort"` to non-zero in the docker `"portMappings"`.
+
```

Comment:
What about:

Since multiple tasks may run on the same node, host ports are assigned dynamically per task unless you use `"requirePorts"`, or set the `"hostPort"` to non-zero in the Docker `"portMappings"`. Note: Docker `"portMappings"` are ignored unless the network type is `BRIDGE`.